NYTimesMostViewedArticles/NetworkClient.swift

```diff
@@ -0,0 +1,78 @@
+//
+//  NetworkClient.swift
+//  NYTimesMostViewedArticles
+//
+//  Created by Sagar Pahwa on 17/11/20.
+//  Copyright © 2020 Sagar Pahwa. All rights reserved.
+//
+
+import Foundation
+
+protocol NetworkClientType {
+    func getMostViewedArticles(period: PeriodSection, completion: @escaping (Response?, Error?)->())
+    
+    func url(period: PeriodSection) -> URL
+}
+
+enum Keys: String {
+    case nyTimesKey = "yxPlqAEPWW3qKi0BgofuzzXeEbxkGfj4"
+    case mockKey = "1234567890"
+}
+
+enum PeriodSection: Int {
+    case day = 1
+    case week = 7
+    case month = 30
+}
+
+struct API {
+    static let baseURL = "http://api.nytimes.com/"
+    static let contentURL = "svc/mostpopular/v2/viewed"
+        //"svc/mostpopular/v2/mostviewed/all-sections/"
+}
+
+
+typealias closure = ()
+
+extension NetworkClientType {
+    func url(period: PeriodSection) -> URL {
+        let string = "\(API.baseURL)/\(API.contentURL)/\(period.rawValue).json?api-key=\(Keys.nyTimesKey.rawValue)"
+        return URL(string: string)!
```

Comment:
Force unwrapping should be avoided as it may lead to crash.

Reply:
yes it is just that using optionals leads to littering of codebase with unwrapping whereas i know the url will exist since it the url format is respected in this case i.e it has a valid scheme, host, path.